### PR TITLE
xhighfive: introducing richer error-stream for runtime_errors

### PR DESCRIPTION
**Pull request by tdegeus**
The new error-stream facilitates debugging.

### CODE REVIEW

File: include/xtensor-io/xhighfive.hpp [hunk 1]
```diff
@@ -25,6 +25,18 @@
 
 namespace xt
 {
+    namespace detail
+    {
+        inline void error(const HighFive::File& file, const std::string &path, std::string message)
```
Comment (wolfv) [inline include/xtensor-io/xhighfive.hpp]: looks good! minor formatting: ampersand at path, and spaces around `+` signs :)

`const std::string& path`

`"path: '" + path ...`

File: include/xtensor-io/xhighfive.hpp [hunk 2]
```diff
@@ -118,7 +130,7 @@ namespace xt
         {
             if (!exist(file, path))
             {
-                throw std::runtime_error("xt::extensions::size: Field does not exist ('"+path+"')");
+                detail::error(file, path, "xt::extensions::size: Field does not exist");
```
Comment (wolfv) [inline include/xtensor-io/xhighfive.hpp]: I wonder if there is a difference of `throw`ing here vs throwing in the helper function? 

`detail::error` could also just return an instance of std::runtime_error that get's thrown at this point here.

File: include/xtensor-io/xhighfive.hpp (hunk 1 above)
Comment (tdegeus) [reply include/xtensor-io/xhighfive.hpp]: Got it!

File: include/xtensor-io/xhighfive.hpp (hunk 2 above)
Comment (tdegeus) [reply include/xtensor-io/xhighfive.hpp]: Not sure either, but it's a good idea.

**Comment by tdegeus** [comment]
That should do it!

BTW upstream the recursive `exist` and `createGroup` are being fixed, I will update as soon as everything is through.

**Comment by wolfv** [comment]
Merging! 

Yep, I saw that they are working on this stuff now. Really cool! :) I guess we'd also need to wait for them to release, but then we should switch to upstream, yes.

**Pull request merged**